src/surge-xt/gui/SurgeGUIEditorValueCallbacks.cpp

```diff
@@ -938,18 +938,24 @@ int32_t SurgeGUIEditor::controlModifierClicked(Surge::GUI::IComponentTagValue *c
             if (lfo_id >= 0)
             {
                 auto msi = modsource_index;
-                contextMenu.addItem(Surge::GUI::toOSCaseForMenu("Rename Modulator"), [this, lfo_id,
-                                                                                      msi, cms]() {
-                    auto mecb = [this, lfo_id, msi](const std::string &nv) {
-                        auto cp =
-                            synth->storage.getPatch().LFOBankLabel[current_scene][lfo_id][msi];
-                        strxcpy(cp, nv.c_str(), CUSTOM_CONTROLLER_LABEL_SIZE);
-                        synth->refresh_editor = true;
-                    };
-                    promptForMiniEdit(
-                        synth->storage.getPatch().LFOBankLabel[current_scene][lfo_id][msi],
-                        "Set Modulator Name", "Rename Modulator", juce::Point<int>(10, 10), mecb);
-                });
+                contextMenu.addItem(
+                    Surge::GUI::toOSCaseForMenu("Rename Modulator..."),
+                    [this, lfo_id, modsource, msi, cms]() {
+                        auto mecb = [this, lfo_id, msi](const std::string &nv) {
+                            auto cp =
+                                synth->storage.getPatch().LFOBankLabel[current_scene][lfo_id][msi];
+                            strxcpy(cp, nv.c_str(), CUSTOM_CONTROLLER_LABEL_SIZE);
+                            synth->refresh_editor = true;
+                        };
+                        promptForMiniEdit(
+                            synth->storage.getPatch().LFOBankLabel[current_scene][lfo_id][msi],
+                            fmt::format("Enter a new name for {:s}:",
+                                        modulatorName(modsource, true)),
```

Comment:
You probably want modusltornamewithinfex here

Reply:
With a small change to its structure, yes. 🙂